src/upgradeable/ERC721CommonUpgradeable.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../refs/IERC721State.sol";
+import "./refs/ERC721NonceUpgradeable.sol";
+import "./ERC721PresetMinterPauserAutoIdCustomizedUpgradeable.sol";
+
+abstract contract ERC721CommonUpgradeable is
+  ERC721NonceUpgradeable,
+  ERC721PresetMinterPauserAutoIdCustomizedUpgradeable,
+  IERC721State
+{
+  error ErrInvalidArrayLength();
+  error ErrNonExistentToken();
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  /**
+   * @inheritdoc IERC721State
+   */
+  function stateOf(uint256 _tokenId) external view virtual override returns (bytes memory) {
+    if (!_exists(_tokenId)) revert ErrNonExistentToken();
+    return abi.encodePacked(ownerOf(_tokenId), nonces[_tokenId], _tokenId);
+  }
+
+  /**
+   * @dev Override `ERC721Upgradeable-_baseURI`.
+   */
+  function _baseURI()
+    internal
+    view
+    virtual
+    override(ERC721Upgradeable, ERC721PresetMinterPauserAutoIdCustomizedUpgradeable)
+    returns (string memory)
+  {
+    return super._baseURI();
+  }
+
+  /**
+   * @dev Override `IERC165-supportsInterface`.
+   */
+  function supportsInterface(bytes4 interfaceId)
+    public
+    view
+    virtual
+    override(ERC721Upgradeable, ERC721PresetMinterPauserAutoIdCustomizedUpgradeable)
+    returns (bool)
+  {
+    return super.supportsInterface(interfaceId);
```

Comment:
`super.method()` only invokes the right-most inherited contract’s method.
Should be something like `interfaceId == type(ERC721Upgradeable).interfaceId || interfaceId == type(ERC721PresetMinterPauserAutoIdCustomizedUpgradeable).interfaceId`

Please write a test for this! Many third parties impl this method incorrectly already.

Reply:
Please check other `super.f()` in other methods as well. Write a few tests!